ui/fuzzy-view.go

```diff
@@ -0,0 +1,126 @@
+// gomuks - A terminal Matrix client written in Go.
+// Copyright (C) 2018 Tulir Asokan
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package ui
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+
+	"github.com/evidlo/fuzzysearch/fuzzy"
+	"maunium.net/go/gomuks/debug"
+	"maunium.net/go/gomuks/matrix/rooms"
+	"maunium.net/go/gomuks/ui/widget"
+	"maunium.net/go/tcell"
+	"maunium.net/go/tview"
+)
+
+type FuzzyView struct {
+	tview.Primitive
+	matches  fuzzy.Ranks
+	selected int
+}
+
+func NewFuzzyView(view *MainView, width int, height int) *FuzzyView {
+
+	rooms := []*rooms.Room{}
+	roomtitles := []string{}
```

Comment:
These should be renamed to something like `roomList` and `roomTitles` so the first one wouldn't conflict with imports and so the second one would use camelCase.